addons/account/wizard/account_payment_register.py

```diff
@@ -492,15 +492,19 @@ def _get_total_amount_in_wizard_currency_to_full_reconcile(self, batch_result, e
             ), False
         elif self.source_currency_id == comp_curr and self.currency_id != comp_curr:
             # Company currency on source line but a foreign currency one on the opposite line.
-            return abs(sum(
-                comp_curr._convert(
-                    aml.amount_residual,
-                    self.currency_id,
-                    self.company_id,
-                    aml.date,
-                )
-                for aml in batch_result['lines']
-            )), False
+            residual_amount = 0.0
+            for aml in batch_result['lines']:
+                if not aml.move_id.payment_id and not aml.move_id.statement_line_id:
+                    conversion_date = self.payment_date
+                else:
+                    conversion_date = aml.date
+            residual_amount += comp_curr._convert(
+                aml.amount_residual,
+                self.currency_id,
+                self.company_id,
+                conversion_date,
+            )
```

Comment:
```suggestion
                residual_amount += comp_curr._convert(
                    aml.amount_residual,
                    self.currency_id,
                    self.company_id,
                    conversion_date,
                )
```
Shouldn't it be inside the for loop?